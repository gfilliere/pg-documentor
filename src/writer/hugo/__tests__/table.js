import { movieTable } from '../../../fixtures/sampleSchema';
import generateTable from '../table';

jest.mock('fs');

describe('generateTable', () => {
  const fs = require('fs'); // eslint-disable-line global-require
  beforeAll(() => {
    fs.mkdirSync('/memory');
    fs.mkdirSync('/memory/hugo');
    fs.mkdirSync('/memory/hugo/schema');
  });

  test('a file for the table is created', async () => {
    expect.assertions(1);
    await generateTable(movieTable, '/memory/hugo/schema');

    expect(
      fs.readFileSync('/memory/hugo/schema/movie.md', 'utf8'),
    ).toMatchSnapshot();
  });
});
