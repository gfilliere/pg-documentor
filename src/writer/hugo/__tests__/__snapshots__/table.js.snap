// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTable a file for the table is created 1`] = `
"+++
title = \\"movie\\"
description = \\"Structure of table movie\\"
+++

Column | Type | Modifiers
----|----|----
active | boolean | not null default false
data | jsonb | default '{}'::jsonb
id | bigint | not null
gid | global_id | not null
title | text | not null
id_poster | bigint | not null

Indexes:
- __movie_pkey__: CREATE UNIQUE INDEX movie_pkey ON movie.movie USING btree (id)
- __movie_movie_gid_idx__: CREATE INDEX movie_movie_gid_idx ON movie.movie USING btree (gid)

Foreign-key constraints:
- __movie_id_poster_fkey__: FOREIGN KEY (id_poster) REFERENCES media.image(id)

Referenced by:
- TABLE movie.movie_has_image CONSTRAINT __movie_has_image_id_movie_fkey__: FOREIGN KEY (id_movie) REFERENCES movie.movie(id)

Triggers:
- __movie_movie_compute_gid__ : CREATE TRIGGER movie_movie_compute_gid BEFORE INSERT OR UPDATE ON movie.movie FOR EACH ROW EXECUTE PROCEDURE abstract.compute_gid()

Inherits:
- abstract.element
"
`;
